carla/data/load_scm/distributions.py

```diff
@@ -0,0 +1,154 @@
+        plt.ylabel("Count")
+        plt.title(fr"Histogram of {self.name}")
+        plt.grid(True)
+        plt.show()
+
+
+class Normal(BaseDistribution):
+    def __init__(self, mean: Union[int, float], var: Union[int, float]):
+        name = f"Normal(mean={mean}, var={var})"
+        super().__init__(name)
+        self.dist = norm(mean, np.sqrt(var))
+
+    def sample(self, size=1) -> Union[np.ndarray, float]:
+        s = self.dist.rvs(size)
+        # returns a single number rather then a list if a single sample
+        return s[0] if size == 1 else s
+
+    def pdf(self, value):
+        return self.dist.pdf(value)
+
+
+class MixtureOfGaussians(BaseDistribution):
+    def __init__(self, probs, means, vars):
+        if not sum(probs) == 1:
+            raise ValueError("Mixture probabilities must sum to 1.")
+        if not len(probs) == len(means) == len(vars):
+            raise ValueError("Length mismatch.")
+
+        name = f"MoG(probs={probs}, means={means}, vars={vars})"
+        super().__init__(name)
+
+        self.probs = probs
+        self.means = means
+        self.vars = vars
+
+    def sample(self, size=1) -> Union[np.ndarray, float]:
+        mixtures = np.random.choice(len(self.probs), size=size, p=self.probs)
+        s = [
+            np.random.normal(self.means[mixture_idx], np.sqrt(self.vars[mixture_idx]))
+            for mixture_idx in mixtures
+        ]
+        return s[0] if size == 1 else np.array(s)
+
+    def pdf(self, value):
+        return np.sum(
+            [
+                prob * norm(mean, var).pdf(value)
+                for (prob, mean, var) in zip(self.probs, self.means, self.vars)
+            ]
+        )
+
+
+class Uniform(BaseDistribution):
+    def __init__(self, lower: Union[int, float], upper: Union[int, float]):
+        if not lower < upper:
+            raise ValueError("upper lower then lower")
+
+        name = f"Uniform(lower={lower}, upper={upper})"
+        super().__init__(name)
+
+        self.lower = lower
+        self.upper = upper
+
+    def sample(self, size=1) -> Union[np.ndarray, float]:
+        s = np.random.uniform(self.lower, self.upper, size=size)
+        return s[0] if size == 1 else s
+
+    def pdf(self, value):
+        raise 1 / (self.upper - self.lower)
+
+
+class Bernoulli(BaseDistribution):
+    def __init__(self, prob: Union[int, float], btype="01"):
+        if not 0 <= prob <= 1:
+            raise ValueError("prob not in correct range")
+
+        name = f"Bernoulli(prob={prob})"
+        super().__init__(name)
+
+        self.prob = prob
+        self.btype = btype  # '01' is standard, '-11' also supported here
+
+    def sample(self, size=1) -> Union[np.ndarray, int]:
+        s = bernoulli.rvs(self.prob, size=size)
+        if self.btype == "-11":
+            s = s * 2 - 1
+        return s[0] if size == 1 else s
+
+    def pdf(self, value):
+        raise Exception("not supported yet; code should not come here.")
```

Comment:
Does the Bernoulli distribution work for SCMs if the pdf method is not supported?